src/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -116,4 +120,32 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
         }
     }
 
+    /***
+    * @description Creates/modifies schedule record(s) after insert/update of recurring donation
+    * @param rds Trigger.new records
+    * @param oldRds Trigger.old records
+    * @param triggerAction BeforeInsert, AfterInsert, etc.
+    */
+    private void handleRDSchedules(List<SObject> rds, List<SObject> oldRds, TDTM_Runnable.Action triggerAction) {
+
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
+            buildNewSchedules(rds);
+        }
+    }
+
+    /***
+    * @description Handles schedule creation after insert of recurring donation
+    * @param rds Trigger.new records
+    */
+    private void buildNewSchedules(List<npe03__Recurring_Donation__c> rds) {
+
+        RD2_ScheduleService scheduleService = new RD2_ScheduleService();
+
+        for (npe03__Recurring_Donation__c rd: rds) {
+            if (rd.Status__c != RD2_Constants.STATUS_CLOSED) {
```

Comment:
I think this check should be in the actual `getNewSchedules()` method rather than here. I added a comment in that method as well.